src/kakarot/instructions/environmental_information.cairo

```diff
@@ -485,14 +486,8 @@ namespace EnvironmentalInformation {
             dst, account.code_len, account.code
         );
 
-        let (implementation) = Kakarot_cairo1_helpers_class_hash.read();
-        let (code_hash) = ICairo1Helpers.library_call_keccak(
-            class_hash=implementation,
-            words_len=dst_len,
-            words=dst,
-            last_input_word=last_word,
-            last_input_num_bytes=last_word_num_bytes,
-        );
+        let (starknet_address) = Kakarot_evm_to_starknet_address.read(evm_address);
+        let (code_hash) = IAccount.get_code_hash(contract_address=starknet_address);
```

Comment:
```suggestion
        let code_hash = account.code_hash
 ```

I don't think a starknet call should be done here, the account and its bytecode should have been cached already at this point